### PR TITLE
Feature/ignore redirect

### PR DESCRIPTION
**Pull request by nicktate**
* Adds delete route for `/cluster` to trigger the stoppage of all containership agents on cluster. Assumes the agents have been run inside docker container with `on-failure` set as the restart policy so they don't relaunch after exiting with a success code.

* Also add in a special header to bypass controlling leader redirect if needed

@normanjoyner

### CODE REVIEW

File: handlers/v1/cluster.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
+'use strict';
+
+const request = require('request');
```
Comment (normanjoyner) [inline handlers/v1/cluster.js]: unused

File: package.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "body-parser": "^1.4.3",
     "express": "^4.2.0",
     "lodash": "^2.4.1",
-    "mocha": "^2.3.4"
+    "mocha": "^2.3.4",
+    "request": "^2.74.0"
```
Comment (normanjoyner) [inline package.json]: unused

File: handlers/v1/cluster.js [hunk 3]
```diff
@@ -11,7 +15,33 @@ module.exports = function(core){
             }
 
             return next();
+        },
+
+        // kill containership on cluster
+        delete: function(req, res, next) {
+            // if not the controlling leader, ignore request (X-Containership-Api-Redirect: false) must have been set
+            if (!core.cluster.praetor.is_controlling_leader()) {
+                res.stash.code = 400;
```
Comment (normanjoyner) [inline handlers/v1/cluster.js]: can we add a response body that indicates they are required to hit the controlling leader (request without `X-ContainerShip-Api-Redirect=false`) in this case?

Comment (nicktate) [inline handlers/v1/cluster.js]: updated outside of diff

**Pull request merged**